test/type_checker_test.dart

```diff
@@ -31,33 +31,35 @@ void main() {
   TypeChecker staticGeneratorForAnnotationChecker;
 
   setUpAll(() async {
-    var resolverDone = new Completer();
-    final resolver = await resolveSource(r'''
+    LibraryElement core;
+    LibraryElement collection;
+    LibraryReader sourceGen;
+    await resolveSource(r'''
       export 'package:source_gen/source_gen.dart';
-    ''', tearDown: resolverDone.future);
+    ''', (resolver) async {
+      core = await resolver.findLibraryByName('dart.core');
+      collection = await resolver.findLibraryByName('dart.collection');
+      sourceGen =
```

Comment:
This is a case where we could still use `tearDown` if we want to keep the variables `final` - I'm open to either approach.